src/realm/query_engine.cpp

```diff
@@ -17,9 +17,65 @@
  **************************************************************************/
 
 #include <realm/query_engine.hpp>
+#include <realm/column_table.hpp>
 
 using namespace realm;
 
+namespace realm {
+
+Columns<SubTable>::Columns(const Columns& other, QueryNodeHandoverPatches* patches)
```

Comment:
It looks like you moved this back into the header file instead. Why not introduce the .cpp file?